File: src/main/java/ordilov/lissn/like/application/LikeFacade.java
package ordilov.lissn.like.application;

import lombok.RequiredArgsConstructor;
import ordilov.lissn.like.domain.LikeCommand;
import ordilov.lissn.like.domain.LikeInfo.LikedPlaylistInfo;
import ordilov.lissn.like.domain.LikeInfo.LikedTrackInfo;
import ordilov.lissn.like.domain.LikeService;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class LikeFacade {

  private final LikeService likeService;

  public LikedPlaylistInfo like(LikeCommand.LikePlaylistRequest request) {
    return likeService.like(request);
  }

  public LikedTrackInfo like(LikeCommand.LikeTrackRequest request) {
    return likeService.like(request);
  }
}

File: src/main/java/ordilov/lissn/like/application/LikeFacade.kt
package ordilov.lissn.like.application

import ordilov.lissn.like.domain.*
import org.springframework.stereotype.Service

@Service
class LikeFacade(
    val likeService: LikeService
) {

    fun like(command: LikePlaylistCommand): LikedPlaylistInfo {
        return likeService.like(command)
    }

    fun like(command: LikeTrackCommand): LikedTrackInfo {
        return likeService.like(command);
    }
}